### PR TITLE
build: fix `cargo publish` by symlinking cli/docs->docs

### PR DESCRIPTION
**Pull request by martinvonz**
# Checklist

If applicable:
- [ ] I have updated `CHANGELOG.md`
- [ ] I have updated the documentation (README.md, docs/, demos/)
- [ ] I have updated the config schema (cli/src/config-schema.json)
- [ ] I have added tests to cover my changes

### CODE REVIEW

**Comment by thoughtpolice** [comment]
FWIW, my other comment on #4630 was meant to say we should:

1. Write a `build.rs` that copies the docs directory to `$OUT_DIR`, which is a variable provided to the build script: https://doc.rust-lang.org/cargo/reference/environment-variables.html#environment-variables-cargo-sets-for-build-scripts
2. Then, change the `include_str!()` call to instead do something like:

```rust
const DOCS_DIR = concat!(env!("OUT_DIR"), "/docs")
```

since `$OUT_DIR` will also be set during the build process, too.

However, I'm still not sure this will solve the problem, because I don't know if the `.crate` itself will get those files. So maybe we do have to symlink things which... Ugh. Well, maybe it's all we can do...

**Comment by thoughtpolice** [comment]
I also wonder if there's a way to test `cargo publish` before we actually merge a PR...

**Comment by martinvonz** [comment]
> However, I'm still not sure this will solve the problem, because I don't know if the `.crate` itself will get those files. So maybe we do have to symlink things which... Ugh. Well, maybe it's all we can do...

Yes, I suspect so.

> I also wonder if there's a way to test `cargo publish` before we actually merge a PR...

I tested it with `cargo publish --dry-run`, which fails with lots of compilation errors because the jj-cli points to the published 0.22.0 version of jj-lib, but it also seems to show that the `include_str!` does not work. The `include_str!` error goes away with this PR.

File: cli/build.rs [hunk 1]
```diff
@@ -37,6 +38,15 @@ fn main() {
     } else {
         println!("cargo:rustc-env=JJ_VERSION={version}");
     }
+
+    let manifest_dir = PathBuf::from(std::env::var("CARGO_MANIFEST_DIR").unwrap());
+    let docs_symlink_path = manifest_dir.join("docs");
```
Comment (yuja) [inline cli/build.rs:43]: nit: maybe we can assume that cwd is `CARGO_MANIFEST_DIR`? `GIT_HEAD_PATH` already relies on that.

**Review by yuja — APPROVED** [review]
Thanks!

File: cli/build.rs (hunk 1 above)
Comment (martinvonz) [reply cli/build.rs]: Good point that we already rely on that. Done.

**Pull request merged**